propulsion-cosmos-reactor/Program.fs

```diff
@@ -0,0 +1,154 @@
+module ReactorTemplate.Program
+
+open Serilog
+open System
+
+exception MissingArg of message : string with override this.Message = this.message
+
+type Configuration(tryGet) =
+
+    let get key =
+        match tryGet key with
+        | Some value -> value
+        | None -> raise (MissingArg (sprintf "Missing Argument/Environment Variable %s" key))
+
+    member _.CosmosConnection =                 get "EQUINOX_COSMOS_CONNECTION"
+    member _.CosmosDatabase =                   get "EQUINOX_COSMOS_DATABASE"
+    member _.CosmosContainer =                  get "EQUINOX_COSMOS_CONTAINER"
+
+module Args =
+
+    open Argu
+    [<NoEquality; NoComparison>]
+    type Parameters =
+        | [<AltCommandLine "-g"; Mandatory>]    ConsumerGroupName of string
+        | [<AltCommandLine "-r"; Unique>]       MaxReadAhead of int
+        | [<AltCommandLine "-w"; Unique>]       MaxWriters of int
+        | [<AltCommandLine "-V"; Unique>]       Verbose
+        | [<AltCommandLine "-C"; Unique>]       CfpVerbose
+        | [<CliPrefix(CliPrefix.None); Unique(*ExactlyOnce is not supported*); Last>] Cosmos of ParseResults<CosmosSourceParameters>
+        interface IArgParserTemplate with
+            member a.Usage = a |> function
+                | ConsumerGroupName _ ->        "Projector consumer group name."
+                | MaxReadAhead _ ->             "maximum number of batches to let processing get ahead of completion. Default: 16."
```

Comment:
2 is probably a better default, except for extremely high throughput scenarios

Reply:
Got it, updated locally with the new template we spoke offline about

Reply:
It's still 16, no?

Reply:
(ah, I see, its just not pushed yet)